src/modules/storage/LedgerStorage.ts

```diff
@@ -1696,8 +1872,9 @@ export class LedgerStorage extends Storages {
                             block.txs[tx_idx]
                         );
 
-                        if (block.txs[tx_idx].payload.length > 0)
-                            await save_payload(this, block.merkle_tree[tx_idx], block.txs[tx_idx]);
+                        if (block.txs[tx_idx].payload.length > 0) {
+                            await putPayload(this, block.header.height, block.merkle_tree[tx_idx], block.txs[tx_idx]);
```

Comment:
I want to separate this process from the block data puting DB transaction process.
The reason is that block collection can be stopped if processes that are prone to errors are grouped together into DB transactions.

https://github.com/bosagora/stoa/blob/62c57ea93ef2598a38b2ff5e98cce12ecdf80998/src/modules/storage/LedgerStorage.ts#L411-L423
Please do it after this db commit.

But the `save_payload`function can be executed within a dB transaction.

Reply:
Done